Duplicate identifier in buffer and send packages

Using the allowJs compiler option, if [buffer](https://github.com/feross/buffer) or [send](https://github.com/pillarjs/send) are dependencies, Duplicate identifier errors are reported:
```log
node_modules/buffer/index.js(94,5): error TS2300: Duplicate identifier 'length'.
node_modules/buffer/index.js(95,5): error TS2300: Duplicate identifier 'parent'.
node_modules/buffer/index.js(229,20): error TS2300: Duplicate identifier 'length'.
node_modules/buffer/index.js(230,20): error TS2300: Duplicate identifier 'parent'.
node_modules/send/index.js(203,22): error TS2300: Duplicate identifier 'root'.
node_modules/send/index.js(212,22): error TS2300: Duplicate identifier 'root'.
```

It seems similar to #6645, but here in addition to this member assignment, prototypes are in play.

--- a/src/compiler/binder.ts
+++ b/src/compiler/binder.ts
@@ -1465,6 +1465,7 @@ namespace ts {
             }
 
             // Declare the method/property
+            // It's acceptable for multiple prototype property assignments of the same identifier to occur
             declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes & ~SymbolFlags.Property);
         }
 

--- a/src/compiler/binder.ts
+++ b/src/compiler/binder.ts
@@ -1465,7 +1465,7 @@ namespace ts {
             }
 
             // Declare the method/property
-            declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes);
+            declareSymbol(funcSymbol.members, funcSymbol, leftSideOfAssignment, SymbolFlags.Property, SymbolFlags.PropertyExcludes & ~SymbolFlags.Property);
         }
 
         function bindCallExpression(node: CallExpression) {

--- a/tests/cases/fourslash/jsDocClasses1.ts
+++ b/tests/cases/fourslash/jsDocClasses1.ts
@@ -1,0 +1,28 @@
+///<reference path="fourslash.ts" />
+
+// @allowNonTsExtensions: true
+// @Filename: Foo.js
+
+//// function fn() {
+//// 	this.foo = 10;
+//// }
+//// fn.prototype.foo = 14;
+//// var x = new fn();
+//// 
+//// function fn2() {
+//// 	this.foo = 10;
+//// 	this.foo = 10;
+//// }
+//// fn2.prototype.foo = 14;
+//// fn2.prototype.foo = 14;
+//// var y = new fn2();
+//// 
+//// function fn3() {
+//// 	this.foo = 10;
+//// }
+//// fn3.prototype.foo = 14;
+//// fn3.prototype.foo = 14;
+//// var z = new fn3();
+
+verify.numberOfErrorsInCurrentFile(0);
+

